### PR TITLE
Switch monkeypatch fixture to yield syntax

### PR DESCRIPTION
**Pull request by jeffwidman**
If you're open to merging this, let me know if I should target a different branch than master and I'll amend the commit to include updates to changelog/authors

### CODE REVIEW

**Comment by nicoddemus** [comment]
Hi @jeffwidman,

Is there a reason for this other than just improving the internal code a bit? If there are not external changes I don't think we should add a `CHANGELOG` entry as this won't affect users.

Otherwise it is a welcome mini-improvement IMO. 😁

File: _pytest/monkeypatch.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def monkeypatch(request):
     will be raised if the set/deletion operation has no target.
     """
     mpatch = MonkeyPatch()
-    request.addfinalizer(mpatch.undo)
-    return mpatch
+    yield mpatch
+    mpatch.undo
```
Comment (The-Compiler) [inline _pytest/monkeypatch.py]: This should be `.undo()` as we actually want to call the method now, not only pass a callable.

**Comment by jeffwidman** [comment]
Purely internal code cleanup. Agreed there's no point in changelog. Fixed and force-pushed

**Comment by nicoddemus** [comment]
Thanks @jeffwidman, appreciate it!

**Pull request merged**